code/game/objects/items/blink_drive.dm

```diff
@@ -0,0 +1,169 @@
+
+	for(var/action in user.actions)
+		if(!istype(action, /datum/action/item_action))
+			continue
+		var/datum/action/item_action/iaction = action
+		if(iaction?.holder_item == src)
+			iaction.set_toggle(FALSE)
+
+///Check if we can use the blink drive and give feedback to the user
+/obj/item/blink_drive/proc/can_use(datum/source, atom/A)
+	SIGNAL_HANDLER
+	var/mob/living/carbon/human/human_user = usr
+	if(human_user.incapacitated() || human_user.lying_angle)
+		return
+	if(is_mainship_level(human_user.z))
+		human_user.balloon_alert(human_user, "can't use here")
+		return
+	if(charges <= 0)
+		human_user.balloon_alert(human_user, "no charge")
+		playsound(src, 'sound/items/blink_empty.ogg', 25, 1)
+		return
+	INVOKE_ASYNC(src, .proc/teleport, A, human_user)
+
+///Handles the actual teleportation
+/obj/item/blink_drive/proc/teleport(atom/A, mob/user)
+	var/turf/target_turf = get_turf(A)
+
+	if(target_turf == user.loc)
+		return
+
+	var/target_distance = get_dist(user, target_turf)
+
+	if(target_distance > BLINK_DRIVE_RANGE)
+		user.balloon_alert(user, "too far")
+		return
+
+	user.face_atom(target_turf)
+
+	var/instability = 0 //certain factors can make the teleport unreliable
+	if(target_distance > BLINK_DRIVE_RANGE - 2)
+		instability ++
+	if(!COOLDOWN_CHECK(src, blink_stability_cooldown))
+		instability ++
+	if(!line_of_sight(user, target_turf, 9))
+		instability ++
+
+	target_turf = pick(RANGE_TURFS(instability, target_turf))
+
+	var/atom/movable/pulled_target = user.pulling
+	if(pulled_target)
+		if(!do_after(user, 0.5 SECONDS, TRUE, user, BUSY_ICON_HOSTILE))
+			return
+		user.balloon_alert(user, "pulled someone through")
+
+	teleport_debuff_aoe(user)
+	user.forceMove(target_turf)
+	if(pulled_target)
+		pulled_target.forceMove(target_turf)
+	teleport_debuff_aoe(user)
+
+	if(target_turf.density || isspaceturf(target_turf))
+		user.emote("gored")
+		user.gib() //telegibbed
+		if(pulled_target && ismob(pulled_target))
+			var/mob/mob_target = pulled_target
+			mob_target.emote("gored")
+			mob_target.gib()
+		return
+
+	COOLDOWN_START(src, blink_stability_cooldown, 1 SECONDS)
+	charges --
+	deltimer(charge_timer)
+	charge_timer = addtimer(CALLBACK(src, .proc/recharge), BLINK_DRIVE_CHARGE_TIME * 2, TIMER_STOPPABLE)
+	update_icon()
+
+///Recharges the drive, and sets another timer if not maxed out
+/obj/item/blink_drive/proc/recharge()
+	charges ++
+	playsound(src, 'sound/items/blink_recharge.ogg', 25, 1)
+	if(charges < BLINK_DRIVE_MAX_CHARGES)
+		charge_timer = addtimer(CALLBACK(src, .proc/recharge), BLINK_DRIVE_CHARGE_TIME, TIMER_STOPPABLE)
+	else
+		charge_timer = null
+	update_icon()
+
+///The effects applied on teleporting from or to a location
+/obj/item/blink_drive/proc/teleport_debuff_aoe(atom/movable/teleporter)
+	playsound(teleporter, 'sound/effects/EMPulse.ogg', 25, 1)
+
+	new /obj/effect/temp_visual/blink_drive(get_turf(teleporter))
+
+	for(var/mob/living/living_target in range(1, teleporter))
+		living_target.adjust_stagger(1)
```

Comment:
okay do you know what orange is

Reply:
today I learnt that center is not Center